homeassistant/components/mqtt/__init__.py

```diff
@@ -220,21 +230,23 @@ def publish_template(hass, topic, payload_template, qos=None, retain=None):
 
 @asyncio.coroutine
 @bind_hass
-def async_subscribe(hass, topic, msg_callback, qos=DEFAULT_QOS,
-                    encoding='utf-8'):
+def async_subscribe(hass: HomeAssistantType, topic: str,
+                    msg_callback: MessageCallbackType,
+                    qos: int = DEFAULT_QOS,
+                    encoding: str = 'utf-8'):
     """Subscribe to an MQTT topic.
 
     Call the return value to unsubscribe.
     """
-    async_remove = \
-        yield from hass.data[DATA_MQTT].async_subscribe(topic, msg_callback,
-                                                        qos, encoding)
+    async_remove = yield from hass.data[DATA_MQTT].async_subscribe(
+        topic, msg_callback, qos, encoding)
     return async_remove
 
 
 @bind_hass
-def subscribe(hass, topic, msg_callback, qos=DEFAULT_QOS,
-              encoding='utf-8'):
+def subscribe(hass: HomeAssistantType, topic: str,
```

Comment:
This method now returns False to indicate a failed state instead of a mixture of False and None.